src/spikeinterface/core/recording_tools.py

```diff
@@ -46,41 +47,39 @@ def get_random_data_chunks(
     # Should be done by changing kwargs with total_num_chunks=XXX and total_duration=YYYY
     # And randomize the number of chunk per segment weighted by segment duration
 
-    # check chunk size
-    num_segments = recording.get_num_segments()
-    for segment_index in range(num_segments):
-        if chunk_size > recording.get_num_frames(segment_index) - 2 * margin_frames:
-            error_message = (
-                f"chunk_size is greater than the number "
-                f"of samples for segment index {segment_index}. "
-                f"Use a smaller chunk_size!"
-            )
-            raise ValueError(error_message)
+    from .node_pipeline import run_traces_pipeline
+    from .job_tools import divide_segment_into_chunks
 
     rng = np.random.default_rng(seed)
-    chunk_list = []
+    all_chunks = []
+
+    ### What seems to be a correct implementation, with no overlapping chunks. Cons:
+    ### margin is not taken into account, and we can have less samples than what is requested
+    # for segment_index in range(recording.get_num_segments()):
+    #     num_frames = recording.get_num_samples(segment_index)
+    #     chunks = divide_segment_into_chunks(num_frames, chunk_size)
+    #     indices = np.random.permutation(np.arange(len(chunks)))[:num_chunks_per_segment]
+    #     for i in indices:
+    #         frame_start, frame_stop = chunks[i]
+    #         all_chunks += [(segment_index, frame_start, frame_stop)]
+
+    ### Former implementation. Cons: overlapping chunks can be drawn
     low = margin_frames
-    size = num_chunks_per_segment
-    for segment_index in range(num_segments):
+    for segment_index in range(recording.get_num_segments()):
         num_frames = recording.get_num_frames(segment_index)
         high = num_frames - chunk_size - margin_frames
-        random_starts = rng.integers(low=low, high=high, size=size)
-        segment_trace_chunk = [
-            recording.get_traces(
-                start_frame=start_frame,
-                end_frame=(start_frame + chunk_size),
-                segment_index=segment_index,
-                return_scaled=return_scaled,
-            )
-            for start_frame in random_starts
-        ]
-
-        chunk_list.extend(segment_trace_chunk)
+        random_starts = rng.integers(low=low, high=high, size=num_chunks_per_segment)
+        for start_frame in random_starts:
+            all_chunks += [(segment_index, start_frame, start_frame + chunk_size)]
 
+    data = run_traces_pipeline(
+        recording, job_kwargs, all_chunks=all_chunks, return_scaled=return_scaled, squeeze_output=True
+    )
     if concatenated:
-        return np.concatenate(chunk_list, axis=0)
+        return data
     else:
-        return chunk_list
+        num_chunks = recording.get_num_segments() * num_chunks_per_segment
+        return np.split(data, chunk_size * np.arange(1, num_chunks))
```

Comment:
@yger you should add `{}` in the docstring and reformat the docs using `_shared_job_kwargs_doc`

See for example the [`extract_waveforms`](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L1690) implementation